### ISSUE TITLE
Support of W3C's accessibility standards

### ISSUE THREAD

**Issue by hokkanent** [KIProtect/klaro#100]
To accomodate users that use assistive software such as screen readers –how does
Klaro consent banner supports W3C's accessibility standards WCAG 2.1 (level AA) and WAI-ARIA?

**Comment by jaller94** [comment]
Klaro has not been tested for accessibility.
We have some plans to make it more accessibly though and will support contributions to make Klaro more accessible.

Klaro has been forked by Empreinte Digitale. The fork [Orejime 🍪](https://github.com/empreinte-digitale/orejime#readme) focuses on accessibility. It follows Web Content Accessibility Guidelines (WCAG) via the French RGAA. I would love [to apply most of there changes to our codebase as well](https://github.com/empreinte-digitale/orejime/issues/6), but it's not trivial and will take some planning and development time.

In regards to screen readers, I'm only familiar with the usage of the ChromeVox screen reader and I know that their detection of semantic HTML and aria features vary, but not how. A list of found issues per screen reader would help to address these. When in doubt, I would use the semantic HTML, hoping for screen readers to improve over time.

**Comment by darylknight** [comment]
I'd love to see this too. I'm currently stuck between a rock and a hard place... between "complying with international cookie laws" and "complying with international accessibility laws".

**Comment by julianeba** [comment]
@jaller94 I have a short list from a specialized agency. That's all the information I have. I hope it helps. 

- The accordion element in the settings of the cookie overlay "↓ 1 service" is a show-hide element and should be a button with aria-label.

- The link "Let me choose", with which the overlay for the cookie settings can be opened, should be a button with aria-label that opens a dialog.

- The modal window itself is not implemented correctly - especially when it comes to [focus management](https://adrianroselli.com/2020/10/dialog-focus-in-screen-readers.html).

**Comment by mrleblanc101** [comment]
@jaller94 $ years later, any news to share ? There hasn't seem to be any progress.

**Comment by darylknight** [comment]
FWIW, I've moved elsewhere. The last commit to this repository was a year ago. The subscription side hasn't worked for a year, adn the support team have stopped replying.
